fix: validate typescript declarations using tests

jkodu opened the pull request:
- Added additional test case for typescript definitions/ interface changes.
- Run typescript definition tests as a part of npm test.

Closes issue: 
- https://github.com/dequelabs/axe-core/issues/979

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: << Name here >> @stephenmathieson

stephenmathieson commented on package.json:
```diff
@@ -55,8 +55,9 @@
-    "build": "grunt",
-    "test": "grunt test",
+		"build": "grunt",
+		"test-dts": "tsc --noImplicitAny typings/axe-core/axe-core-tests.ts",
```
We should disable other language features too (implicit this, etc).

Also, these options are better stored in `tsconfig.json`, as the CLI flags will get out of hand quickly.

jkodu replied:
agreed, moved to tsconfig.json

stephenmathieson commented on typings/axe-core/axe-core-tests.ts:
```diff
@@ -1,73 +1,105 @@
 
-var context:any = document
-var $fixture:any = {}
+const context:any = document;
```
Looks like this file should be run thru Prettier 😉

stephenmathieson commented:
It's super difficult to see what changed here. A brief glance just shows whitespace changes, `s/var/const` and moving to arrow functions. Were there any important changes made to the `*.ts` files?

jkodu commented:
@stephenmathieson 
I reverted and reworked the actual test enhancement into this commit - https://github.com/dequelabs/axe-core/pull/1016/commits/0f7583c02cca34abbbba40ca32b6cde73e8a7c19#diff-6420a0bcddb269c3e42adec97308bab9

The rest of the commits are adding `ts` files to prettier and some cleanup by introducing tsconfig.

Review when you get a mo. Cheers.

stephenmathieson commented:
Hey Jey! First off, it was awesome to meet you in person! Let's make sure we do that again sometime :)

Now... this is still pretty difficult to review. IMO this should have been two different PRs (one for style stuff, one for actual functionality). Not a huge deal tho.

stephenmathieson commented on package.json:
```diff
@@ -56,14 +56,15 @@
   "scripts": {
     "build": "grunt",
-    "test": "grunt test",
+		"test-dts": "tsc",
```
Weird indent here.

jkodu commented:
@stephenmathieson - can you fill out the reviewer details in the PR please. Thanks

stephenmathieson commented:
Right. Yes. Done.
